src/main/java/com/hotels/hcommon/ssh/session/DelegatingSessionFactory.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (C) 2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.hcommon.ssh.session;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.jcraft.jsch.JSchException;
+import com.jcraft.jsch.Proxy;
+import com.jcraft.jsch.Session;
+import com.pastdev.jsch.SessionFactory;
+
+class DelegatingSessionFactory implements SessionFactory {
+  private static final Logger log = LoggerFactory.getLogger(DelegatingSessionFactory.class);
+
+  final /* VisibleForTesting */ SessionFactory delegate;
```

Comment:
The comment position looks a bit weird here, can it not go before "final"? Also, doesn't it need an @ for an annotation or is this just a plain comment for devs to know?

Reply:
Do they need to be visible since you can create this class and pass whatever you like in via the constructor?

Reply:
this project is Guava-free. We do check the values in a couple of tests

Reply:
OK, just move the comment up a level